tests/security/389ds/tls_389ds_server.pm

```diff
@@ -0,0 +1,112 @@
+    zypper_call("in 389-ds");
+    assert_script_run("wget --quiet " . data_url("389ds/instance.inf") . " -O /tmp/instance.inf");
+    assert_script_run("dscreate from-file /tmp/instance.inf");
+    validate_script_output("dsctl localhost status", sub { m/Instance.*is running/ });
+
+    # Configure CA Certificates for TLS
+    assert_script_run("wget --quiet " . data_url("389ds/tls_cert.sh") . " -O /tmp/tls_cert.sh");
+    assert_script_run("wget --quiet " . data_url("389ds/.dsrc") . " -O /root/.dsrc");
+    assert_script_run("bash /tmp/tls_cert.sh $local_name $ca_dir");
+    assert_script_run("certutil -D -d $inst_ca_dir -n Server-Cert");
+    assert_script_run("certutil -D -d $inst_ca_dir -n Self-Signed-CA");
+    assert_script_run("dsctl localhost tls import-server-key-cert $ca_dir/server.pem $ca_dir/server.key");
+    assert_script_run("dsctl localhost tls import-ca $ca_dir/myca.pem myca");
+    assert_script_run("cp $ca_dir/myca.pem $inst_ca_dir/ca.crt");
+    systemctl("restart dirsrv\@localhost.service");
+
+    # Configure host names for C/S communication
+    assert_script_run("sed -i '/$local_ip.*/d' /etc/hosts");
+    assert_script_run("sed -i '/$remote_ip.*/d' /etc/hosts");
+    assert_script_run("echo '$local_ip $local_name.example.com $local_name' >> /etc/hosts");
+    assert_script_run("echo '$remote_ip $remote_name.example.com $remote_name' >> /etc/hosts");
+    assert_script_run("hostname $local_name");
+
+    # Create ldap user and group,generate sample sssd.conf file
+    my $ldap_user    = 'wilber';
+    my $ldap_passwd  = 'pw_389ds';
+    my $ldap_group   = 'server_admins';
+    my $uid          = '1003';
+    my $gid          = '1003';
+    my $display_name = 'Wilber Fox';
+    assert_script_run(
+"dsidm localhost user create --uid $ldap_user --cn $ldap_user --displayName '$display_name' --uidNumber $uid --gidNumber $gid --homeDirectory /home/$ldap_user"
+    );
+    script_run_interactive(
+        "dsidm localhost account reset_password uid=$ldap_user,ou=people,dc=example,dc=com",
+        [
+            {
+                prompt => qr/Enter new password.*/m,
+                string => "$ldap_passwd\n",
+            },
+            {
+                prompt => qr/CONFIRM.*/m,
+                string => "$ldap_passwd\n",
+            },
+        ],
+        60
+    );
+    script_run_interactive(
+        "dsidm localhost group create",
+        [
+            {
+                prompt => qr/Enter value.*/m,
+                string => "$ldap_group\n",
+            },
+        ],
+        60
+    );
+    assert_script_run("dsidm localhost group add_member $ldap_group uid=$ldap_user,ou=people,dc=example,dc=com");
+    assert_script_run("dsidm localhost client_config sssd.conf $ldap_group > /tmp/sssd.conf");
+    assert_script_run("sed -i '1,2d' /tmp/sssd.conf");
+    assert_script_run("sed -i 's/^ldap_uri =.*\$/ldap_uri = ldaps:\\/\\/$local_name.example.com/' /tmp/sssd.conf");
```

Comment:
Can you please describe those lines too?

Reply:
Sure, will do